lib/strverscmp.c

```diff
@@ -0,0 +1,62 @@
+/* Compare strings while treating digits characters numerically.
+ * musl as a whole is licensed under the following standard MIT license:
+ * ----------------------------------------------------------------------
+ * Copyright © 2005-2020 Rich Felker, et al.
+
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+ * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ * ----------------------------------------------------------------------
+ */
+
+#if defined(OPEN_BSD) || defined(__NetBSD__)
+#include <ctype.h>
+#include <string.h>
+
+int strverscmp(const char *l0, const char *r0)
```

Comment:
We already have `frr_version_cmp()` in defaults.c doing the same thing pretty much, please use that

Reply:
ok. nice.

Reply:
> ok. nice.

actually, it will not work. I come back to strverscmp proposal.
```
NAM1 vs NAM2: before / PR11146initial / frr_version_cmp / strcmp / strverscmp
foo vs bar: 4 / 4 / 1 / 4 / 4
foo-001 vs bar-001: 4 / 4 / 1 / 4 / 4
foo-001 vs foo-1: 0 / -1 / 0 / -1 / -1
foo-1 vs foo-001: 0 / 1 / 0 / 1 / 1
foo-100 vs foo-1: 1 / 1 / 1 / 48 / 1
foo-1_bar-01 vs foo-1_bar-1: 0 / -1 / 0 / -1 / -1
foo-1_bar-01 vs foo-1_bar-10: -1 / -1 / -1 / -1 / -1
foo-1_bar-1 vs foo-1_bar-10: -1 / -1 / -1 / -48 / -1
foo-1_bar-1 vs foo-1_bar-1: 0 / 0 / 0 / 0 / 0
foo1 vs foo2: -1 / -1 / -1 / -1 / -1
foo2 vs foo10: -1 / -1 / -1 / 1 / -1
foo10 vs foo11: -1 / -1 / -1 / -1 / -1
123foo vs 123bar: 4 / 4 / 1 / 4 / 4
007 vs 7: 0 / -7 / 0 / -7 / -7
007 vs 0: 1 / 1 / 1 / 48 / -1
foo-007 vs foo-0: 1 / 1 / 1 / 48 / -1
007-foo vs 7-foo: 0 / -7 / 0 / -7 / -7
foo4000000000 vs foo4000000000: 0 / 0 / 0 / 0 / 0
foo4000000000 vs foo04000000000: 0 / 4 / 0 / 4 / 4
r1-eth0 vs r1-eth1: -1 / -1 / -1 / -1 / -1
```